educe/learning/edu_input_format.py

```diff
@@ -28,13 +31,16 @@ def _dump_edu_input_file(docs, f):
             edu_txt = edu.text().replace('\n', ' ')
             # subgroup: sentence identifier, backoff on EDU id
             sent_idx = edu2sent[i]
-            subgroup = ('{}_sent{}'.format(grouping, sent_idx)
-                        if sent_idx is not None
-                        else edu_gid)
+            if sent_idx is None:
+                subgroup = edu_gid
+            elif isinstance(sent_idx, six.string_types):
```

Comment:
Hah, I was starting to mull over the idea of maybe edu2sent being some kind of function, which I think sounds a bit like what you're saying.  

Sure, thanks! Well, I'm not in a terrible terrible hurry at the moment (still am working on the messy scikit-stac branch which will need to be rebased on top of this if/when it becomes STAC), but if you'd prefer having this off the mental todo list (and on to the larger design one) I'm happy with the approach.